caas/kubernetes/provider/k8s_test.go

```diff
@@ -276,19 +313,42 @@ func (s *K8sBrokerSuite) TestDestroy(c *gc.C) {
 	ctrl := s.setupBroker(c)
 	defer ctrl.Finish()
 
-	// Delete operations below return a not found to ensure it's treated as a no-op.
+	ns := &core.Namespace{ObjectMeta: v1.ObjectMeta{Name: "test"}}
+	namespaceWatcher := s.k8sNewFakeWatcher()
+
 	gomock.InOrder(
+		s.mockNamespaces.EXPECT().Watch(
+			v1.ListOptions{
+				FieldSelector:        fields.OneTermEqualSelector("metadata.name", "test").String(),
+				IncludeUninitialized: true,
+			},
+		).
+			Return(namespaceWatcher, nil),
 		s.mockNamespaces.EXPECT().Delete("test", s.deleteOptions(v1.DeletePropagationForeground)).Times(1).
-			Return(s.k8sNotFoundError()),
+			Return(nil),
 		s.mockStorageClass.EXPECT().DeleteCollection(
 			s.deleteOptions(v1.DeletePropagationForeground),
 			v1.ListOptions{LabelSelector: "juju-model==test"},
 		).Times(1).
 			Return(s.k8sNotFoundError()),
+		s.mockNamespaces.EXPECT().Get("test", v1.GetOptions{IncludeUninitialized: true}).Times(1).
+			Return(nil, errors.New("any non k8s not found error")),
+		s.mockNamespaces.EXPECT().Get("test", v1.GetOptions{IncludeUninitialized: true}).Times(1).
+			Return(nil, s.k8sNotFoundError()),
 	)
 
+	go func(w *watch.RaceFreeFakeWatcher, clk *testclock.Clock) {
+		for _, f := range []func(runtime.Object){w.Add, w.Modify, w.Delete} {
+			if !w.IsStopped() {
+				clk.WaitAdvance(time.Second, testing.LongWait, 1)
+				f(ns)
+			}
+		}
+	}(namespaceWatcher, s.clock)
+
 	err := s.broker.Destroy(context.NewCloudCallContext())
 	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(workertest.CheckKilled(c, s.watcher), gc.ErrorMatches, "bad catacomb Kill: tomb.ErrDying")
```

Comment:
Normally we expect CheckKilled to test for a nil error.
"bad catacomb kill" doesn't look right

Reply:
yes, you are right. just fixed a bug in `kubernetesWatcher.loop`. now it's nil